### DISCUSSION TITLE
💾 Add Export/Import Option (Backup)

### DISCUSSION THREAD

**Discussion by matchcom-cahors** [LeoKlaus/Paperparrot#126] [Feature Requests]
When using the App without a paperless-ngx server, it should be possible to export all files at once. All files should be exported as one Zip. It would be nice if the metadata could also be exported in this process. Additionally, it would be helpful if the import of the data were possible so that it can be used as a backup and restore feature.

**Comment by LeoKlaus** [comment]
I really like this idea and have planned to add something like this for quite some time now. I can't give any timeframe or estimate for this, but it's coming.

**Comment by LeoKlaus** [comment]
Export is already functional, import will come later.

**Comment by MaximilianHerrmann** [comment]
Hallo Mr Klaus,
do you have an estimate for the export function? I have a little issue with changing servers and a failed backup, so that the files downloaded to my telephone are de facto an backup of some documents.
The server has the docs in the Database, but the files are missing (I guess that was a problem during migrating the server container from one proxmox environment to another - so no bug/problem with paperless/paperparrot).

Easiest way would be, if your export-feature would be available - as far as I can see, that is not yet implemeted.

Will the app maintain the filenames used in the server database and export the files that way? Or is there the possibility to setup the export naming that way?

Maybe it would be also easily possible to grant access to the sandboxed documents folder, when the phone is connected to the mac. My DebekaGesundheits App for example gives access to the app folders.

Best regards,
Max Herrmann

**Reply by LeoKlaus** [reply]
You can just call me Leo :)

The export function is implemented but not available when connected to a server. I can start showing it with the next update, though.
If you downloaded the original files, they're stored under their original name and exported as such.

> Maybe it would be also easily possible to grant access to the sandboxed documents folder, when the phone is connected to the mac. My DebekaGesundheits App for example gives access to the app folders.

It is possible and fairly easy. The problem is that Apple requires apps that do this to handle all interactions with that folder, which is much more complicated. I don't intend to add any folder consumption or similar features to Paperparrot, so I don't see any reason to enable access to the documents folder other than your specific case.
The existing backup option already exports the entire folder, so there's really no need to add this.

If you send me a short mail at support@paperparrot.me, I can send you an invite for the TestFlight beta of Paperparrot. That version already includes the export option when connected to a server.
